core/src/main/java/org/hibernate/ogm/dialect/spi/BaseGridDialect.java

```diff
@@ -44,4 +46,19 @@ public boolean supportsSequences() {
 	public DuplicateInsertPreventionStrategy getDuplicateInsertPreventionStrategy(EntityKeyMetadata entityKeyMetadata) {
 		return DuplicateInsertPreventionStrategy.LOOK_UP;
 	}
+
+	@Override
+	public void forEachTuple(ModelConsumer consumer, EntityKeyMetadata... entityKeyMetadatas) {
+		for (EntityKeyMetadata entityKeyMetadata : entityKeyMetadatas) {
+			List<Tuple> tuples = getTuples(entityKeyMetadata);
+			for (Tuple tuple : tuples) {
+				consumer.consume(tuple);
+			}
+		}
+	}
+
+	// Should be abstract, but high impact on all subclasses
+	protected List<Tuple> getTuples(EntityKeyMetadata entityKeyMetadata) {
```

Comment:
Note that we are introducing a different `getTuples` method as part of a different improvement. Our version receives a list of `EntityKey` to load.